index.js

```diff
@@ -1,300 +1,147 @@
+  FALSE: Buffer.from([0]),
+}
+const NIL = {
+  TF: Buffer.from([6, 2]),
+}
+NIL.TFD = NIL.TF
 
 const encoder = {
-  feed(feedId) {
-    let tf
-    if (feedId.endsWith('.ed25519')) tf = CLASSIC_FEED_TF
-    else if (feedId.endsWith('.ggfeed-v1')) tf = GABBYGR_FEED_TF
-    else if (feedId.endsWith('.bbfeed-v1')) tf = BENDYBT_FEED_TF
-    else throw new Error('Unknown feed format: ' + feedId)
-
-    const dotIndex = feedId.lastIndexOf('.')
-    const b64part = feedId.substring(1, dotIndex)
-    const d = Buffer.from(b64part, 'base64')
-
-    return Buffer.concat([tf, d])
+  boolean(input) {
+    const d = input ? BOOL.TRUE : BOOL.FALSE
+    return Buffer.concat([BOOL.TF, d])
   },
+  classic(input, type, format) {
+    let data = input
+    if (type.sigil) data = data.slice(1)
+    if (format.suffix) data = data.slice(0, -format.suffix.length)
 
-  message(msgId) {
-    let tf
-    if (msgId.endsWith('.sha256')) tf = CLASSIC_MSG_TF
-    else if (msgId.endsWith('.ggmsg-v1')) tf = GABBYGR_MSG_TF
-    else if (msgId.endsWith('.bbmsg-v1')) tf = BENDYBT_MSG_TF
-    else throw new Error('Unknown msg ID: ' + msgId)
-
-    const dotIndex = msgId.lastIndexOf('.')
-    const b64part = msgId.substring(1, dotIndex)
-    const d = Buffer.from(b64part, 'base64')
-
-    return Buffer.concat([tf, d])
+    return Buffer.concat([type.code, format.code, Buffer.from(data, 'base64')])
   },
-
-  blob(blobId) {
-    let tf
-    if (blobId.endsWith('.sha256')) tf = CLASSIC_BLOB_TF
-    else throw new Error('Unknown blob ID: ' + blobId)
-
-    const dotIndex = blobId.lastIndexOf('.')
-    const b64part = blobId.substring(1, dotIndex)
-    const d = Buffer.from(b64part, 'base64')
-
-    return Buffer.concat([tf, d])
+  string(input) {
+    return Buffer.concat([Buffer.from([6, 0]), Buffer.from(input, 'utf8')])
   },
+}
 
-  box(boxedStr) {
-    if (boxedStr.endsWith('.box')) {
-      const b64part = boxedStr.substring(0, boxedStr.length - '.box'.length)
-      const d = Buffer.from(b64part, 'base64')
-      return Buffer.concat([BOX1_TF, d])
-    } else if (boxedStr.endsWith('.box2')) {
-      const b64part = boxedStr.substring(0, boxedStr.length - '.box2'.length)
-      const d = Buffer.from(b64part, 'base64')
-      return Buffer.concat([BOX2_TF, d])
-    } else throw new Error('Unknown boxed string: ' + boxedStr)
-  },
+function encode(input) {
+  /* cases we don't encode */
+  if (input === undefined || Buffer.isBuffer(input) || Number.isInteger(input))
+    return input
 
-  signature(sig) {
-    const b64part = sig.substring(0, sig.length - '.sig.ed25519'.length)
-    const d = Buffer.from(b64part, 'base64')
-    return Buffer.concat([SIGNATURE_TF, d])
-  },
+  if (typeof input === 'string') {
+    /* looks for classic sigil/suffix matches */
+    const { type, format } = findClassicTypeFormat(input, TYPES)
+    if (type) {
+      if (format) return encoder.classic(input, type, format)
+      else throw new Error(`Unknown ${type.type} format`)
```

Comment:
```suggestion
      else throw new Error(`Unknown ${format} format`)
```

Reply:
Don't agree.
This is meant to output e.g. "Unknown feed format".  The type.type = "feed" here.